### PR TITLE
--[Bugfix/Refactor]-PBR Skin Rendering Prep/Bugfixes.

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR addresses the following issues/refactors : 
- Some common Drawable functionality shared between GenericDrawables (Phong/Flat) and PBRDrawables was promoted to the base Drawable class. This includes management for lights, shader keys, and skin data and handling, in order to reduce needless code duplication and also as pre-requisite work for the upcoming PBR Skin rendering support.
- ~~The PBR shaders were originally implemented to perform calculations in model space, while the Phong and Flat shaders work in Camera space (ModelView transform). This caused the PBR shader to perform many redundant matrix multiplications unnecessarily.  This distinction has been removed, and now PBR operates in Camera space, just like Phong and Flat shaders.~~ This change is temporarily reverted until a reasonable solution to calculating IBL properly in camera space can be derived.
- The PBR shaders were originally able to handle double-sided textures but this was disabled due to an issue with rendering artifacts being produced at edges. Since the PBR rewrite, this issue seems to have been addressed, and so this PR reactivates the ability to disable backface culling for double-sided textures.

## How Has This Been Tested
All c++ and python tests pass locally

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/gfx/Drawable.h [hunk 1]
```diff
@@ -175,13 +187,63 @@ class Drawable : public Magnum::SceneGraph::Drawable3D {
...
+   */
+  void buildSkinJointTransforms();
+
+  /**
+   * @brief Update lighting-related parameters on every draw call
+   * @tparam ShaderType is the type of shader being passed (wrapped in a
+   * Magnum::Resource).
+   * @param transformationMatrix The transformation matrix passed to this
+   * drawables draw function.
+   * @param camera The camera passsed to this drawable's draw function
```
Comment (0mdc) [inline src/esp/gfx/Drawable.h:220]: Nit: `passsed`

File: src/esp/gfx/Drawable.h [hunk 2]
```diff
@@ -190,6 +252,52 @@ class Drawable : public Magnum::SceneGraph::Drawable3D {
...
+  std::vector<float> lightRanges(lightSetup_->size(), dummyRange);
+
+  // std::vector<Mn::Color3> lightSpecularColors;
+  // lightSpecularColors.reserve(lightSetup_->size());
+
+  for (Mn::UnsignedInt i = 0; i < lightSetup_->size(); ++i) {
+    const auto& lightInfo = (*lightSetup_)[i];
+    Mn::Vector4 pos = getLightPositionRelativeToCamera(
+        lightInfo, transformationMatrix, cameraMatrix);
+    // flip directional lights to faciliate faster, non-forking calc in
```
Comment (0mdc) [inline src/esp/gfx/Drawable.h:277]: Nit: `faciliate`

File: src/esp/gfx/Drawable.cpp [hunk 3]
```diff
@@ -41,5 +46,44 @@ DrawableGroup* Drawable::drawables() {
...
+  }
+
+}  // Drawable::buildSkinJointTransforms
+
+void Drawable::resizeJointTransformArray(Mn::UnsignedInt jointCount) {
+  if (jointTransformations_.size() != jointCount) {
+    Corrade::Containers::arrayResize(jointTransformations_, jointCount);
+  }
+}
+
```
Comment (0mdc) [inline src/esp/gfx/Drawable.cpp:87]: 👍

**Review by 0mdc — APPROVED** [review]
LGTM! Left some very minor comments.

**Review by 0mdc — CHANGES_REQUESTED** [review]
As discussed, there is a transform issue when projecting the environment map.

For reference, see the mirror in this video.

https://github.com/facebookresearch/habitat-sim/assets/110583667/7529a2dc-2b74-4fbd-b421-63a04670d0d4

**Comment by jturner65** [comment]
> As discussed, there is a transform issue when projecting the environment map.
> 
> For reference, see the mirror in this video.
> pbr_test-2023-10-02_11.50.20.mp4

I'm reverting the changes to the PBR shader using camera space vs world space until this issue with IBl being calculated in camera space can be addressed properly.

**Review by 0mdc — APPROVED** [review]
LGTM! The issue is fixed with the latest iteration.

**Pull request merged**